controlpanel/api/aws.py

```diff
@@ -259,9 +261,9 @@ def statement(self, sid):
     def add_resource(self, arn, sid):
         statement = self.statement(sid)
         if statement:
-            resources = statement.setdefault('Resource', [])
-            if arn not in resources:
-                resources.append(arn)
+            statement['Resource'] = statement.get('Resource', [])
```

Comment:
I kind of prefer this to the `setdefault` to be honest, the "get or use default" is a more common pattern IMHO and it's probably less surprising as you see what you get if `Resource` is not there at the point of the `get()` call.

But is this still needed if you add the `Resource` keys?

Reply:
Yes. This is defensive coding in case the statement with the specified Sid exists, but has no Resource key. Unlikely but possible.

Reply:
`setdefault` is part of the core language and shouldn't be surprising, but I agree that this is probably clearer.